app/scripts/lib/url.js

```diff
@@ -89,6 +89,16 @@ function (_) {
       return origin;
     },
 
+    /**
+     * Returns true if given "uri" has HTTP or HTTPS scheme so it is navigable
+     *
+     * @param {String} uri
+     * @returns {boolean}
+     */
+    isNavigable: function (uri) {
+      return /^(http|https):\/\//.test(uri);
```

Comment:
We could have some minimal validation here– maybe just a check that extra characters follow the protocol. We wouldn't want to be too strict and block valid URLs.